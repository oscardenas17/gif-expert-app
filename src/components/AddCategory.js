import React, { useState } from 'react'

export const AddCategory = () => {

    const [inputValue, setinputValue] = useState('hola mundo');

    const handleInputChange = (e) =>{
        // console.log(e.target.value);
        setinputValue(e.target.value);
    }

    const handleSubmit = (e) =>{
        e.preventDefault();
    }

    return (
        <form onSubmit = {handleSubmit}>
           <input
                 type="text"
                 value = {inputValue}
                 onChange = {handleInputChange}

           />
        </form>
    )
}
