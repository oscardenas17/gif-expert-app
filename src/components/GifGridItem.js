import React from 'react'

export const GifGridItem = ( {id,title, url} ) => {
    // console.log( id, title, url);

    return (
        <div>
          <img src= {url} alt={title} className="source" />
          <p>{title}</p>
        </div>
    )
}
