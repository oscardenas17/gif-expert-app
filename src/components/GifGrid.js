import React from 'react'
import PropTypes from 'prop-types';

import { useFetchGifs } from '../hooks/useFetchGifs'
import { GifGridItem } from './GifGridItem';

export const GifGrid = ( {category}) => {
  
    // const state = useFetchGifs(); // hook
    const {data: images, loading} = useFetchGifs(category);

    return (
        <>
            <h3 className="animate__fadeOut animate__delay-2s">{category}</h3>
         
            {loading && <p className="animate_animated animate_flash animate__delay-5s">Loading</p>}
            <div className="card-grid">
                
                {
                    images.map( img => (
                        <GifGridItem  
                                key={img.id}
                                // img= {img} />
                                {...img} 
                            />
                    ))
                }
                
            
            </div>            
         </> 

    )
}

GifGrid.propTypes  = {
    category: PropTypes.string.isRequired
}

 