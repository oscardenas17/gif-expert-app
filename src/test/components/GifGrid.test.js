import { shallow } from 'enzyme';
import React from 'react';
import { GifGrid } from "../../components/GifGrid";
import { useFetchGifs } from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs')

describe('Pruebas en <GifGrid />', () => {

    const category = 'Vegeta'

    test('debe mostrarse ok', () => {
         
        
        useFetchGifs.mockReturnValue({
            data: [],
            loading: true
        });

        const wrapper = shallow (<GifGrid  category={category} /> );

        expect(wrapper).toMatchSnapshot();

    });


    test('debe mostrar items cuando cargan imganes de useFetchGifs', () => {
        
        const gifs = [{
            id: 'ABC',
            url: 'https://localhost/cualquier/cosa.jpg',
            title: 'cualquier cosa'
        }]
        
        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false
        });
        
        const wrapper = shallow (<GifGrid  category={category} /> );

        expect(wrapper).toMatchSnapshot()
;       

    });

})