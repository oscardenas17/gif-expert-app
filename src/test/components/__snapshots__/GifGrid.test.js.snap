// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifGrid /> debe mostrar items cuando cargan imganes de useFetchGifs 1`] = `
<Fragment>
  <h3
    className="animate__fadeOut animate__delay-2s"
  >
    Vegeta
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="ABC"
      key="ABC"
      title="cualquier cosa"
      url="https://localhost/cualquier/cosa.jpg"
    />
  </div>
</Fragment>
`;

exports[`Pruebas en <GifGrid /> debe mostrarse ok 1`] = `
<Fragment>
  <h3
    className="animate__fadeOut animate__delay-2s"
  >
    Vegeta
  </h3>
  <p
    className="animate_animated animate_flash animate__delay-5s"
  >
    Loading
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
